app/unit-tests/cose_test.cpp

```diff
@@ -22,7 +22,123 @@ using namespace testutils;
 
 namespace
 {
-  TEST(CoseTest, GetHeaders)
+  TEST(CoseTest, DecodeAllProtectedHeaders)
+  {
+    auto phdr_b = create_valid_protected_header_bytes();
+
+    QCBORDecodeContext ctx;
+    QCBORDecode_Init(&ctx, cbor::from_bytes(phdr_b), QCBOR_DECODE_MODE_NORMAL);
+    EXPECT_NO_THROW(cose::decode_protected_header(ctx));
+  }
+
+  // add a test case to use payloads from test/payloads directory
+  // NOLINTBEGIN(bugprone-unchecked-optional-access)
+  TEST(CoseTest, DecodeTSSHeaders)
+  {
+    std::string filepath = "test_payloads/css-attested-cosesign1-20250617.cose";
+    std::ifstream file(filepath, std::ios::binary);
+    ASSERT_TRUE(file.is_open());
+
+    // Get file size
+    file.seekg(0, std::ios::end);
```

Comment:
There's a more readable `std::filesystem::file_size()`, but it takes a path and so leads to a double open, ymmv.

Reply:
yep it is easier to read